### PR TITLE
Added S3 User credentials Auth functionality

### PR DESCRIPTION
**Pull request by Anish-Saxena**
Taking Issue #7 into consideration, I ventured on to implement option for S3 User credential based auth functionality in this elasticsearch-cron repo. I have also made the PR for required changes in elasticsearch-operator repo.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -45,11 +45,15 @@ var (
 	argElasticURL = ""
 	argUsername   = ""
 	argPassword   = ""
+	argRepoAccessKey = ""
+	argRepoSecretKey = ""
+	argRepoRegion = "eu-west-1"
```
Comment (stevesloka) [inline main.go]: I think we shouldn't have a default here, it causes issues for folks who don't set but operate in different regions.

File: main.go [hunk 2]
```diff
@@ -45,11 +45,15 @@ var (
...
 	argUseSSL     = true
+	argUseRepoAuth = false
```
Comment (stevesloka) [inline main.go]: I think we could remove this key and just use the existence of `argRepoAcessKey` & `argRepoSecretKey` to determine if we want to do an authenticated request.

File: main.go (hunk 1 above)
Comment (Anish-Saxena) [reply main.go]: Right, I have removed the default here.

File: main.go (hunk 2 above)
Comment (Anish-Saxena) [reply main.go]: The UseRepoAuth key has been removed and the Authenticated request now depends on existence of (both) Access and Secret keys.

**Pull request merged**